ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/layout.tsx

```diff
@@ -10,20 +12,30 @@ type RootLayoutProps = {
   children: React.ReactNode;
 };
 
-export default function RootLayout({ params: { environmentSlug }, children }: RootLayoutProps) {
+export default async function RootLayout({
+  params: { environmentSlug },
+  children,
+}: RootLayoutProps) {
+  const { env, envs } = await getEnvs(environmentSlug);
```

Comment:
Here I fetch the environments server side and pass them into the env context provider so the provider itself does not have to fetch the data after render client side.